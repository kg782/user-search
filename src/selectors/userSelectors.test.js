import {
  allUsersSelector,
  administratorSelector,
  archivedSelector,
  favoritesSelector,
  nonAdminSelector
} from './userSelectors';

const users = [
  {
    id: 0,
    fullName: 'Suzy Cuningham',
    email: 'suzy.cuningham@gmail.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: true,
    favorite: true,
    archived: false
  },
  {
    id: 1,
    fullName: 'Bobby Daniels',
    email: 'bobbyD@outlook.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: true,
    favorite: false,
    archived: false
  },
  {
    id: 2,
    fullName: 'John Walker',
    email: 'johnnyWalker@blue.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: true,
    favorite: false,
    archived: false
  },
  {
    id: 3,
    fullName: 'Eddy Stevens',
    email: 'eStevens@yahoo.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: false,
    favorite: true,
    archived: false
  },
  {
    id: 4,
    fullName: 'Jan Williams',
    email: 'jDubz@msn.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: false,
    favorite: false,
    archived: false
  },
  {
    id: 5,
    fullName: 'John Doe',
    email: 'john@msn.com',
    date: 'Oct 1, 2015',
    lastLogin: '1 year ago',
    administrator: true,
    favorite: true,
    archived: true
  }
];

test('usersSelector filters all user except archived', () => {
  expect(allUsersSelector(users)).toMatchSnapshot();
});

test('administratorSelector filters', () => {
  expect(administratorSelector(users)).toMatchSnapshot();
});

test('archivedSelector filters', () => {
  expect(archivedSelector(users)).toMatchSnapshot();
});

test('favoritesSelectorfilters ', () => {
  expect(favoritesSelector(users)).toMatchSnapshot();
});

test('nonAdminSelector filters', () => {
  expect(nonAdminSelector(users)).toMatchSnapshot();
});
