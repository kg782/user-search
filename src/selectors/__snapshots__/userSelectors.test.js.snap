// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`administratorSelector filters 1`] = `
Array [
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "suzy.cuningham@gmail.com",
    "favorite": true,
    "fullName": "Suzy Cuningham",
    "id": 0,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "bobbyD@outlook.com",
    "favorite": false,
    "fullName": "Bobby Daniels",
    "id": 1,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "johnnyWalker@blue.com",
    "favorite": false,
    "fullName": "John Walker",
    "id": 2,
    "lastLogin": "1 year ago",
  },
]
`;

exports[`archivedSelector filters 1`] = `
Array [
  Object {
    "administrator": true,
    "archived": true,
    "date": "Oct 1, 2015",
    "email": "john@msn.com",
    "favorite": true,
    "fullName": "John Doe",
    "id": 5,
    "lastLogin": "1 year ago",
  },
]
`;

exports[`favoritesSelectorfilters  1`] = `
Array [
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "suzy.cuningham@gmail.com",
    "favorite": true,
    "fullName": "Suzy Cuningham",
    "id": 0,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": false,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "eStevens@yahoo.com",
    "favorite": true,
    "fullName": "Eddy Stevens",
    "id": 3,
    "lastLogin": "1 year ago",
  },
]
`;

exports[`nonAdminSelector filters 1`] = `
Array [
  Object {
    "administrator": false,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "eStevens@yahoo.com",
    "favorite": true,
    "fullName": "Eddy Stevens",
    "id": 3,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": false,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "jDubz@msn.com",
    "favorite": false,
    "fullName": "Jan Williams",
    "id": 4,
    "lastLogin": "1 year ago",
  },
]
`;

exports[`usersSelector filters all user except archived 1`] = `
Array [
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "suzy.cuningham@gmail.com",
    "favorite": true,
    "fullName": "Suzy Cuningham",
    "id": 0,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "bobbyD@outlook.com",
    "favorite": false,
    "fullName": "Bobby Daniels",
    "id": 1,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": true,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "johnnyWalker@blue.com",
    "favorite": false,
    "fullName": "John Walker",
    "id": 2,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": false,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "eStevens@yahoo.com",
    "favorite": true,
    "fullName": "Eddy Stevens",
    "id": 3,
    "lastLogin": "1 year ago",
  },
  Object {
    "administrator": false,
    "archived": false,
    "date": "Oct 1, 2015",
    "email": "jDubz@msn.com",
    "favorite": false,
    "fullName": "Jan Williams",
    "id": 4,
    "lastLogin": "1 year ago",
  },
]
`;
